### ISSUE TITLE
how to add toggle button to enable or disable pip mode in exoplayer

### ISSUE THREAD

**Issue by Programmingtimes** [google/ExoPlayer#10161]
i have created app with pip mode. but user is asking to support of external player also they want pip mode to work when user leave. so issue is when user press external player button it leave app and pip mode code get executed also external app play with it . is there any way to manually control pip mode to be turn off or disable.
also i want another specific feature like pip mode can be disable if user dont want via their setting menu.  not requesting any codes. never encourage that. want some input or idea.

**Comment by ojw28** [comment]
It's quite hard to follow what you're asking, but there's a fairly detailed guide about PiP support [here](https://developer.android.com/guide/topics/ui/picture-in-picture). It looks like you have to explicitly call `Activity.enterPictureInPictureMode` to enter PiP mode in the first place, so it's unclear why you couldn't just not call that method in your `Activity` for the case that the user doesn't want PiP functionality to be enabled.

**Comment by ThreeMinutesFix** [comment]
![msg1624374853-3973](https://user-images.githubusercontent.com/103521732/164148479-98bdef46-2366-4874-af2e-2116a88631af.jpg)
this is youtube application where in settings they provided a toggle button to control working of pip mode. i want to implement same function

**Comment by ojw28** [comment]
Creating a toggle button in a settings activity, that persists a boolean preference somehow in your app, seems like a general Android development question that's outside the scope of this issue tracker. You should probably ask on StackOverflow or similar.

Once you have a boolean preference in your app, I think you just need to read its value and enter PiP mode (i.e., by calling `Activity.enterPictureInPictureMode`) conditionally on it being enabled.

**Comment by ThreeMinutesFix** [comment]
thanks for the reply.  i try to implement the idea that you have given.
https://stackoverflow.com/questions/71835676/how-to-control-exoplayer-pip-via-switch-button
. i have already written that in stackoverflow but there is no answer yet so i directly asked here as this forum is active. ( not to misuse this but to get my solution/input immediatly)
